torchtune/data/_messages.py

```diff
@@ -683,18 +714,60 @@ class AlpacaToMessages(Transform):
     due to this custom logic.
 
     Args:
-        train_on_input (bool): Whether the model is trained on the user prompt or not.
-            Default is True.
+        masking_strategy (Optional[str]): masking strategy to use for model training.
+            Must be one of: `train_on_all`, `train_on_assistant`, `train_on_last`.
+            Default is "train_on_all".
+            - ``train_on_all``: both user and assistant messages are unmasked
+            - ``train_on_assistant``: user messages are masked, only assistant messages are unmasked
+            - ``train_on_last``: only the last assistant message is unmasked
         column_map (Optional[Dict[str, str]]): a mapping to change the expected "instruction", "input",
             and "output" column names to the actual column names in the dataset. Default is None,
             keeping the default column names.
+        train_on_input (Optional[bool]): whether the model is trained on the user prompt or not.
+            Deprecated parameter and will be removed in a future release.
+            Default is None.
+
+    Raises:
+        ValueError:
+            If ``column_map`` is provided and ``instruction`` not in ``column_map``, or
+                ``output`` not in ``column_map``
     """
 
     def __init__(
-        self, train_on_input: bool = True, column_map: Optional[Dict[str, str]] = None
+        self,
+        masking_strategy: Optional[str] = "train_on_all",
+        column_map: Optional[Dict[str, str]] = None,
+        train_on_input: Optional[bool] = None,
     ):
-        self.train_on_input = train_on_input
-        self.column_map = column_map
+        if train_on_input is not None:
```

Comment:
I thought we had a deprecated param utility, but it looks like it never got reviewed: https://github.com/pytorch/torchtune/pull/2321. should make a note there to also replace this code if this PR merges first @ebsmothers